tests/api.c

```diff
@@ -33606,12 +33607,16 @@ static int test_wolfSSL_IMPLEMENT_ASN1_FUNCTIONS(void)
     XFREE(der, NULL, DYNAMIC_TYPE_ASN1);
     DPP_BOOTSTRAPPING_KEY_free(bootstrap);
 
+    /* Test integer */
+    AssertNotNull(test_asn1 = TEST_ASN1_new());
+    der = NULL;
+    AssertIntEQ(i2d_TEST_ASN1(test_asn1, &der), 4);
+    XFREE(der, NULL, DYNAMIC_TYPE_ASN1);
+    TEST_ASN1_free(test_asn1);
+
     /* Test error cases. */
-    AssertNull(TEST_ASN1_new());
```

Comment:
Restore the failure case.
Change it to be an OCTET_STRING now that you have added support for INTEGER.